homeassistant/components/device_tracker/bt_smarthub.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for BT Smart Hub (Sometimes referred to as BT Home Hub 6).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.bt_smarthub/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (DOMAIN, PLATFORM_SCHEMA,
+                                                     DeviceScanner)
+from homeassistant.const import CONF_HOST
+
+REQUIREMENTS = ['btsmarthub_devicelist==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DEFAULT_IP = '192.168.1.254'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=CONF_DEFAULT_IP): cv.string,
+})
+
+
+def get_scanner(hass, config):
+    """Return a BT SmartHub scanner if successful."""
+    scanner = BTSmartHubScanner(config[DOMAIN])
+
+    return scanner if scanner.success_init else None
+
+
+class BTSmartHubScanner(DeviceScanner):
+    """This class queries a BT Smart Hub."""
+
+    def __init__(self, config):
+        """Initialise the scanner."""
+        _LOGGER.info("Initialising BT Smart Hub")
+        self.host = config[CONF_HOST]
+        self.last_results = {}
+
+        # Test the router is accessible
+        data = self.get_bt_smarthub_data()
+        self.success_init = data is not None
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        self._update_info()
+        return [client['mac'] for client in self.last_results]
+
+    def get_device_name(self, device):
+        """Return the name of the given device or None if we don't know."""
+        if not self.last_results:
+            return None
+        for client in self.last_results:
+            if client['mac'] == device:
+                return client['host']
+        return None
+
+    def _update_info(self):
+        """Ensure the information from the BT Home Hub 5 is up to date.
+
+        Return boolean if scanning successful.
```

Comment:
Nothing is checking the return value.

Reply:
Thanks for this. Just to make sure - are these values being read anywhere else in home assistant.
Couldn't find any docs on how to structure a device tracker so just used this as a guide:

https://github.com/home-assistant/home-assistant/blob/46ea28a4f81b121967a4041a5baa5d34e0827ac7/homeassistant/components/device_tracker/swisscom.py

They seem to structure it in this way and I assumed that it might be checked by something

Reply:
No, `_update_info` is a private method so it should only be used locally.